StarRailBattleSim/src/powers/AbstractPower.java

```diff
@@ -135,133 +123,43 @@ public void onRemove() {
 
     }
 
+    /**
+     * Get the value of a stat
+     * @param stat The stat to get
+     * @return The value of the stat, 0 if the stat is not set
+     */
+    @NotNull
     public float getStat(PowerStat stat) {
-        switch (stat) {
-            case FLAT_HP:
-                return this.bonusFlatHP;
-            case HP_PERCENT:
-                return this.bonusHPPercent;
-            case FLAT_ATK:
-                return this.bonusFlatAtk;
-            case ATK_PERCENT:
-                return this.bonusAtkPercent;
-            case FLAT_DEF:
-                return this.bonusFlatDef;
-            case DEF_PERCENT:
-                return this.bonusDefPercent;
-            case FLAT_SPEED:
-                return this.bonusFlatSpeed;
-            case SPEED_PERCENT:
-                return this.bonusSpeedPercent;
-            case SAME_ELEMENT_DAMAGE_BONUS:
-                return this.bonusSameElementDamageBonus;
-            case EFFECT_HIT:
-                return this.bonusEffectHit;
-            case EFFECT_RES:
-                return this.bonusEffectRes;
-            case BREAK_EFFECT:
-                return this.bonusBreakEffect;
-            case HEALING:
-                return this.bonusHealing;
-            case DAMAGE_BONUS:
-                return this.bonusDamageBonus;
-            case DAMAGE_TAKEN:
-                return this.bonusDamageTaken;
-            case DEFENSE_REDUCTION:
-                return this.defenseReduction;
-            case DEFENSE_IGNORE:
-                return this.defenseIgnore;
-            case RES_PEN:
-                return this.resPen;
-            case CRIT_CHANCE:
-                return this.bonusCritChance;
-            case CRIT_DAMAGE:
-                return this.bonusCritDamage;
-            case ENERGY_REGEN:
-                return this.bonusEnergyRegen;
-            case TAUNT_VALUE:
-                return this.bonusTauntValue;
-            case WEAKNESS_BREAK_EFF:
-                return this.bonusWeaknessBreakEff;
-            default:
-                throw new IllegalArgumentException("Unknown stat: " + stat);
-        }
+        return this.stats.getOrDefault(stat, 0f);
     }
 
+    /**
+     * Increase the value of a stat, calls getStat internally
+     * @param stat The stat to increase
+     * @param value The value to increase by
+     */
+    public void increaseStat(PowerStat stat, float value) {
+        this.setStat(stat, this.getStat(stat) + value);
+    }
+
+    /**
+     * Add a value to a stat, calls setStat internally, for chaining.
+     * @param stat The stat to add to
+     * @param value The value to add
+     * @return The power object
+     */
+    public AbstractPower addStat(PowerStat stat, float value) {
```

Comment:
Couldn't `setStat` return `AbstractPower` instead of `void` so you can chain `setStats` together, and then this function becomes unecessary?

Reply:
Ha, yes very true. I added them while adding the Trace stuff, didn't think of it it seems